InvenTree/templates/js/translated/order.js

```diff
@@ -2491,6 +2491,57 @@ function allocateStockToSalesOrder(order_id, line_items, options={}) {
                 },
                 value: options.shipment || null,
                 auto_fill: true,
+                secondary: {
+                    method: 'POST',
+                    title: '{% trans "Add Shipment" %}',
+                    fields: function() {
+                        var ref = null;
+
+                        // TODO: Refactor code for getting next shipment number
+                        inventreeGet(
+                            '{% url "api-so-shipment-list" %}',
```

Comment:
@SchrodingersGat I think an API endpoint for this would be the safer option/ do we already have something that could be used for that?